### ISSUE TITLE
Calling casted function pointer with non-function source yields invalid IR

### ISSUE THREAD

**Issue by MasterQ32** [ziglang/zig#3608]
```zig
pub fn main() void 
{
    const magicAssemblerFunction = @ptrCast(extern fn () void, @intToPtr(*u8,0x1000));
    magicAssemblerFunction();
}
```
will result in
```
Assertion failed at /deps/zig/src/ir.cpp:11300 in ir_resolve_fn. This is a bug in the Zig compiler.
```

Zig Version: 0.5.0+55685ae78

**Comment by Vexu** [comment]
This has been fixed between Zig 0.5.0+55685ae78 and 0.5.0+9bc4f8ea7 and now works correctly.
```zsh
$ zig version
0.5.0+55685ae78
$ zig run a.zig
Assertion failed at /home/vexu/Documents/zig/zig/src/ir.cpp:11300 in ir_resolve_fn. This is a bug in the Zig compiler.
```
```zsh
$ zig version
0.5.0+9bc4f8ea7
$ zig run a.zig
Segmentation fault at address 0x1000
???:?:?: 0x1000 in ??? (c)


/usr/lib/zig/std/special/start.zig:225:22: 0x2272fb in std.special.posixCallMainAndExit (run)
            root.main();
                     ^
/usr/lib/zig/std/special/start.zig:106:5: 0x2271df in std.special._start (run)
    @noInlineCall(posixCallMainAndExit);
    ^
zsh: abort (core dumped)  zig run a.zig
```

**Comment by MasterQ32** [comment]
> This has been fixed between Zig 0.5.0+55685ae78 and 0.5.0+9bc4f8ea7 and now works correctly.

Okay, this is crazy. I did an update to the last CI version from the website and it gets fixed magically in the next hours after that. Still have verify that it works on my machine :slightly_smiling_face:

**Comment by LemonBoy** [comment]
Dupe of #3558, fixed by #3587
